lnode-src/duckcoin.go

```diff
@@ -119,170 +121,4 @@ func main() {
-	if err != nil {
-		http.Error(w, err.Error(), http.StatusInternalServerError)
-		return
-	}
-	w.Write(bytes)
-}
-func handleWriteBlock(w http.ResponseWriter, r *http.Request) {
-	w.Header().Set("Content-Type", "application/json")
-	var b util.Block
-
-	decoder := json.NewDecoder(io.LimitReader(r.Body, 1e6))
-	if err := decoder.Decode(&b); err != nil {
-		fmt.Println("Bad request. This may be caused by a block that is too big (more than 1mb) but these are usually with malicious intent.")
-		respondWithJSON(w, http.StatusBadRequest, r.Body)
-		return
-	}
-	fmt.Println(b)
-	defer r.Body.Close()
-
-	if err := isValid(b, NewestBlock); err == nil {
-		addBlockToChain(b)
-	} else {
-		respondWithJSON(w, http.StatusBadRequest, "Invalid block. "+err.Error())
-		fmt.Println("Rejected block")
-		return
-	}
-	respondWithJSON(w, http.StatusCreated, "Block accepted.")
-}
-func respondWithJSON(w http.ResponseWriter, code int, payload interface{}) {
-	w.Header().Set("Content-Type", "application/json")
-	response, err := json.MarshalIndent(payload, "", "  ")
-	if err != nil {
-		w.WriteHeader(http.StatusInternalServerError)
-		w.Write([]byte("HTTP 500: Internal Server Error: " + err.Error()))
-		return
-	}
-	w.WriteHeader(code)
-	w.Write(response)
-}
-
-func isValid(newBlock, oldBlock util.Block) error {
-	const blockDataLimit = 1e3 * 250
-	const txDataLimit = 1e3 * 250
-
-	if newBlock.Tx.Amount < 0 {
-		return errors.New("Amount is negative")
-	}
-	if oldBlock.Index+1 != newBlock.Index {
-		return errors.New("Index should be " + strconv.FormatInt(oldBlock.Index+1, 10))
-	}
-	if oldBlock.Hash != newBlock.PrevHash {
-		return errors.New("PrevHash should be " + oldBlock.Hash)
-	}
-	if util.CalculateHash(newBlock) != newBlock.Hash {
-		return errors.New("Block Hash is incorrect. This usually happens if your Difficulty is set incorrectly. Restart your miner.")
-	}
-	if !util.IsHashSolution(newBlock.Hash, Difficulty) {
-		return errors.New("Block is not a solution (does not have Difficulty zeros in hash)")
-	}
-	if len(newBlock.Data) > blockDataLimit {
-		return errors.New("Block's Data field is too large. Should be >= 250 kb")
-	}
-	if len(newBlock.Tx.Data) > txDataLimit {
-		return errors.New("Transaction's Data field is too large. Should be >= 250 kb")
-	}
-	if newBlock.Tx.Amount > 0 {
-		if util.DuckToAddress(newBlock.Tx.PubKey) != newBlock.Tx.Sender {
-			return errors.New("Pubkey does not match sender address")
-		}
-		if ok, err := util.CheckSignature(newBlock.Tx.Signature, newBlock.Tx.PubKey, newBlock.Hash); !ok {
-			if err != nil {
-				return err
-			} else {
-				return errors.New("Invalid signature")
-			}
-		}
-		if newBlock.Tx.Sender == newBlock.Solver {
-			if Balances[newBlock.Tx.Sender]+1 < newBlock.Tx.Amount { // plus 1 because that's the reward
-				return errors.New("Insufficient balance")
-			}
-		} else {
-			if Balances[newBlock.Tx.Sender] < newBlock.Tx.Amount {
-				return errors.New("Insufficient balance")
-			}
-		}
-	}
-	return nil
-}
```

Comment:
You shouldn't delete this!

Reply:
(basically just don't change this lower part of the code except the reward)

Reply:
ughhh